frontend/src/pages/room.tsx

```diff
@@ -95,7 +94,7 @@ const Room: React.FC<RouteComponentProps<any> & RoomProps> = (props) => {
             if (res[0]) setInput(newInput);
             else console.log('Failed', body, patch);
         });
-    }, [input]);
+    }, [input, body, dmp]);
```

Comment:
Are you using some extension that auto-fills dependency list for you?
good catches, but here we musn't have body inside, just `input, dmp]`
I just forgot to change console.log here
```suggestion
    }, [input, dmp]);
```